core/domain/exp_domain.py

```diff
@@ -3453,6 +3454,82 @@ def _convert_states_v20_dict_to_v21_dict(cls, states_dict):
                 content_ids_to_audio_translations)
         return states_dict
 
+    @classmethod
+    def _convert_states_v21_dict_to_v22_dict(cls, states_dict):
+        """Converts from version 21 to 22. Version 22 converts all Rich Text
+        Editor content to be compatible with the textAngular format.
+
+        Args:
+            states_dict: dict. A dict where each key-value pair represents,
+                respectively, a state name and a dict used to initialize a
+                State domain object.
+
+        Returns:
+            dict. The converted states_dict.
+        """
+        for state_dict in states_dict.values():
+            state_dict['content']['html'] = (
+                html_cleaner.convert_to_text_angular(
```

Comment:
I thought I wrote this in a previous comment, but I'll mention it again -- we should not be having the nested call all over the place. The call here should be html_cleaner.convert_to_text_angular(state_dict['content']['html']). Any "deeper" validation should be something that convert_to_text_angular() deals with internally.

Reply:
Fixed this! (Sorry I missed that)